ambry-cloud/src/main/java/com/github/ambry/cloud/HelixVcrClusterParticipant.java

```diff
@@ -189,9 +186,7 @@ public void participate() throws Exception {
         cloudConfig.vcrClusterZkConnectString);
     VcrStateModelFactory stateModelFactory = Utils.getObj(cloudConfig.vcrHelixStateModelFactoryClass, this);
     manager.getStateMachineEngine().registerStateModelFactory(stateModelFactory.getStateModelName(), stateModelFactory);
-    if (cloudConfig.cloudContainerCompactionEnabled) {
-      registerContainerDeletionSyncTask(manager.getStateMachineEngine());
-    }
+    registerContainerDeletionSyncTask(manager.getStateMachineEngine());
```

Comment:
What's the reason that we don't need check the "cloudContainerCompactionEnable" flag?

Reply:
This check here is redundant. It is being checked below registerContainerDeletionSyncTask.